RecoTracker/MkFitCore/src/MkBuilder.cc

```diff
@@ -0,0 +1,1427 @@
+                                                         iteration_dir);
+
+          if (layer_plan_it.is_pickup_only() || theEndCand == 0)
+            continue;
+
+          // vectorized loop
+          for (int itrack = 0; itrack < theEndCand; itrack += NN) {
+            int end = std::min(itrack + NN, theEndCand);
+
+            dprint("processing track=" << itrack << ", label="
+                                       << eoccs[seed_cand_idx[itrack].first][seed_cand_idx[itrack].second].label());
+
+            //fixme find a way to deal only with the candidates needed in this thread
+            mkfndr->inputTracksAndHitIdx(eoccs.refCandidates(), seed_cand_idx, itrack, end, false);
+
+            //propagate to layer
+            dcall(pre_prop_print(curr_layer, mkfndr.get()));
+
+            (mkfndr.get()->*fnd_foos.m_propagate_foo)(
+                layer_info.propagate_to(), end - itrack, prop_config.finding_inter_layer_pflags);
+
+            dcall(post_prop_print(curr_layer, mkfndr.get()));
+
+            dprint("now get hit range");
+            mkfndr->selectHitIndices(layer_of_hits, end - itrack);
+
+            find_tracks_handle_missed_layers(
+                mkfndr.get(), layer_info, tmp_cands, seed_cand_idx, region, start_seed, itrack, end);
+
+            dprint("make new candidates");
+            mkfndr->findCandidates(layer_of_hits, tmp_cands, start_seed, end - itrack, fnd_foos);
+
+          }  //end of vectorized loop
+
+          // sort the input candidates
+          for (int is = 0; is < n_seeds; ++is) {
+            dprint("dump seed n " << is << " with N_input_candidates=" << tmp_cands[is].size());
+
+            std::sort(tmp_cands[is].begin(), tmp_cands[is].end(), sortCandByScore);
+          }
+
+          // now fill out the output candidates
+          for (int is = 0; is < n_seeds; ++is) {
+            if (!tmp_cands[is].empty()) {
+              eoccs[start_seed + is].clear();
+
+              // Put good candidates into eoccs, process -2 candidates.
+              int n_placed = 0;
+              bool first_short = true;
+              for (int ii = 0; ii < (int)tmp_cands[is].size() && n_placed < params.maxCandsPerSeed; ++ii) {
+                TrackCand &tc = tmp_cands[is][ii];
+
+                // See if we have an overlap hit available, but only if we have a true hit in this layer
+                // and pT is above the pTCutOverlap
+                if (tc.pT() > params.pTCutOverlap && tc.getLastHitLyr() == curr_layer && tc.getLastHitIdx() >= 0) {
+                  CombCandidate &ccand = eoccs[start_seed + is];
+
+                  HitMatch *hm = ccand[tc.originIndex()].findOverlap(
+                      tc.getLastHitIdx(), layer_of_hits.refHit(tc.getLastHitIdx()).detIDinLayer());
+
+                  if (hm) {
+                    tc.addHitIdx(hm->m_hit_idx, curr_layer, hm->m_chi2);
+                    tc.incOverlapCount();
+
+                    // --- ROOT text tree dump of all found overlaps
```

Comment:
Could be removed?

Reply:
Addressed in Round 4 follow-up PR trackreco/cmssw#79 commit: https://github.com/trackreco/cmssw/pull/79/commits/d081786b240bad6e7d367004a8b05c8de61ad18b